### DISCUSSION TITLE
Custom 404 when using page templates

### DISCUSSION THREAD

**Discussion by jakobjyberg** [gatsbyjs/gatsby#34063] [Help] [answered]
I want to customize the 404 error component when using page templates and createPage but it doesn't seems to work. If I have the default pages folder with 404.js file in there the projects builds without any errors in console but in the browser the page is stuck on a white page with the message "Preparing requested page". 

The reason I want to modify the 404 component is that I want to redirect invalid urls to /, is there any other way to do this?

**Comment by LekoArts** [comment, marked as answer]
The 404 page also receives `location` as a prop, so depending on the route you can return/render different things on the 404 page.

**Reply by jakobjyberg** [reply]
I see but my site doesn't work properly when I add the 404 component when using createPage method. It only works if the page folder is removed (= I cannot modify 404 page). The browser get stuck on "Preparing requested page", but it builds without any errors.

**Reply by jakobjyberg** [reply]
<img width="1440" alt="Skärmavbild 2021-12-01 kl  13 20 35" src="https://user-images.githubusercontent.com/12053299/144233575-a88b806a-3efb-4e62-9f37-3d8a60748a2d.png">
<img width="1440" alt="Skärmavbild 2021-12-01 kl  13 20 47" src="https://user-images.githubusercontent.com/12053299/144233580-5a897bb0-ea9c-4afe-b81f-5e14789d6a7f.png">

As soon as I remove the pages folder everything works, is this a bug? Other pages is generated dynamically.

**Reply by jakobjyberg** [reply]
The issue actually resolved by itself, don't know what caused the issue.
